R/wrap_vfold.R

```diff
@@ -0,0 +1,218 @@
+                                    ...) {
+
+  if (missing(radius) || missing(buffer)) {
+    use_vfold <- NULL
+    if (missing(radius) && missing(buffer)) {
+      use_vfold <- c(i = "Or use `rsample::vfold_cv() to use a non-spatial V-fold")
+    }
+    rlang::abort(
+      c(
+        "`spatial_buffer_vfold_cv()` requires both `radius` and `buffer` be provided",
+        i = "Set either `radius` or `buffer` to NULL to not use that argument",
+        use_vfold
+      )
+
+    )
+  }
+  if (missing(radius)) radius <- NULL
+  if (missing(buffer)) buffer <- NULL
+
+  n <- nrow(data)
+  v <- check_v(v, n, "rows")
+
+  indices <- rsample::vfold_cv(
+    data = data,
+    v = v,
+    repeats = repeats,
+    strata = strata,
+    breaks = breaks,
+    pool = pool,
+    ...
+  )
+
+  cv_att <- list(v = v,
+                 repeats = repeats,
+                 strata = !is.null(strata),
+                 radius = radius,
+                 buffer = buffer)
+
+  if ("sf" %in% class(data)) {
+    rset_class <- c("spatial_buffer_vfold_cv", "spatial_rset", "rset")
+    rsplit_class <- c("spatial_buffer_vfold_split", "spatial_rsplit")
+  } else {
+    rset_class <- c("spatial_buffer_vfold_cv", "rset")
+    rsplit_class <- c("spatial_buffer_vfold_split")
+  }
+
+  wrap_vfold(
+    data = data,
+    indices = indices,
+    rsplit_class = rsplit_class,
+    rset_class = rset_class,
+    radius = radius,
+    buffer = buffer,
+    n = n,
+    v = v,
+    cv_att = cv_att
+  )
+
+}
+
+
+#' @rdname spatial_vfold
+#'
+#' @export
+spatial_leave_location_out_cv <- function(data,
+                                          group,
+                                          v = NULL,
+                                          radius = NULL,
+                                          buffer = NULL,
+                                          ...) {
+
+  if (!missing(group)) {
+    group <- tidyselect::eval_select(rlang::enquo(group), data)
+  }
+
+  if (missing(group) || length(group) == 0) {
+    group <- NULL
+  } else {
+    if (is.null(v)) v <- length(unique(data[[group]]))
+    v <- check_v(v, length(unique(data[[group]])), "locations")
+  }
+
+  n <- nrow(data)
+
+  indices <- rsample::group_vfold_cv(
+    data = data,
+    v = v,
+    group = group,
+    ...
+  )
+
+  cv_att <- list(v = v,
+                 group = group,
+                 radius = radius,
+                 buffer = buffer)
+
+  if ("sf" %in% class(data)) {
+    rset_class <- c("leave_location_out_cv", "spatial_rset", "rset")
+    rsplit_class <- c("leave_location_out_split", "spatial_rsplit")
+  } else {
+    rset_class <- c("leave_location_out_cv", "rset")
+    rsplit_class <- c("leave_location_out_split")
+  }
+
+  wrap_vfold(
+    data = data,
+    indices = indices,
+    rsplit_class = rsplit_class,
+    rset_class = rset_class,
+    radius = radius,
+    buffer = buffer,
+    n = n,
+    v = v,
+    cv_att = cv_att
+  )
+
+}
+
+wrap_vfold <- function(data,
```

Comment:
What would you think about a different name than `wrap_vfold()`? From this name, I was expecting this function to "wrap" i.e. call `vfold_cv()`. What about something like `tweak_vfold()`?

Reply:
How's `edit_rset()`? It doesn't need to be from `vfold_cv()` at all.

Reply:
That sounds great 👍

Reply:
I went with `posthoc_buffer_rset` to _really_ emphasize what's happening (because the only editing done is the buffering), let me know if that's too much :laughing: